src/ImageManipulation.php

```diff
@@ -1057,4 +1062,98 @@ protected function castDimension($value, $dimension)
         }
         return $value;
     }
+
+    private function copyImageBackend()
+    {
+        // Store result in new DBFile instance
+        /** @var DBFile $file */
+        $file = DBField::create_field(
+            'DBFile',
+            [
+                'Filename' => $this->getFilename(),
+                'Hash' => $this->getHash(),
+                'Variant' => $this->getVariant()
+            ]
+        );
+
+        $file->setImageBackend($this->getImageBackend());
+        return $file->setOriginal($this);
+    }
+
+    public function setAttribute($name, $value)
```

Comment:
This overrides the `setAttribute` method for our new trait. The reason I do this, is that we need our Image to be "Immutable" when we manipulate it. This allows us to return a new copy each time.